contractcourt/htlc_timeout_resolver.go

```diff
@@ -160,6 +160,44 @@ func (h *htlcTimeoutResolver) claimCleanUp(
 	return nil, h.Checkpoint(h)
 }
 
+// resolveWithFailure is a helper method that's called once the HTLC output is
+// spent by the remote party without revealing the pre-image (e.g. by a
+// a retribution transaction). It'll resolve the HTLC with a failure on the
+// incoming link.
+func (h *htlcTimeoutResolver) resolveWithFailure() (ContractResolver, error) {
```

Comment:
`godoc` comment and method name should reflect that this path is also used for an HTLC timeout flow, which I guess can be interpreted as a special type of "failure".

Reply:
`emergencyCleanUp` is the best I can come up for the both cases. They're both unusual and require special measures (e.g. failing an incoming link).